test/e2e/performanceprofile/functests/2_performance_update/updating_profile.go

```diff
@@ -643,6 +644,82 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
 		})
 	})
+
+	Context("Offlined CPU API", func() {
+		BeforeEach(func() {
+			By("Saving the old performance profile")
+			initialProfile = profile.DeepCopy()
+		})
+
+		It("[disruptive] should set offline cpus after deploy PAO", func() {
+			for _, node := range workerRTNodes {
+				onlineCPUCount, err := nodes.ExecCommandOnNode([]string{"nproc", "--all"}, &node)
+				Expect(err).ToNot(HaveOccurred())
+				onlineCPUInt, err := strconv.Atoi(onlineCPUCount)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(onlineCPUInt).Should(BeNumerically(">=", 3))
+			}
+
+			// Create new performance with offlined
+			reserved := performancev2.CPUSet("0")
+			isolated := performancev2.CPUSet("1")
+			offlined := performancev2.CPUSet("2-3")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Enable UserLevelNetworking and add Devices in Profile")
+			profile.Spec.CPU = &performancev2.CPU{
+				Reserved: &reserved,
+				Isolated: &isolated,
+				Offlined: &offlined,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+
+			//Check offlined cpus are setting correctly
+			for _, node := range workerRTNodes {
+				Eventually(func() bool {
+					// Check if node is Ready
```

Comment:
need to get an update node object here

Reply:
how?

Reply:
https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/utils/nodes/nodes.go

Reply:
Ok, deleted eventually and add the updated nodes, if "Waiting for MCP being updated" waits correctly, node should be ready and its not worth to check if its actually ready, right?

Reply:
It's a good idea to add the updated node object event if its not necessary (in this case) because the ExecCommandOnNode, only needs node.name and that doesn't change

`// GetMachineConfigDaemonByNode returns the machine-config-daemon pod that runs on the specified node
func GetMachineConfigDaemonByNode(node *corev1.Node) (*corev1.Pod, error) {
	listOptions := &client.ListOptions{
		Namespace:     testutils.NamespaceMachineConfigOperator,
		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}),
		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}),
	}

	mcds := &corev1.PodList{}
	if err := testclient.Client.List(context.TODO(), mcds, listOptions); err != nil {
		return nil, err
	}

	if len(mcds.Items) < 1 {
		return nil, fmt.Errorf("failed to get machine-config-daemon pod for the node %q", node.Name)
	}
	return &mcds.Items[0], nil
}
`

Just to show that it was not the root issue of the bug